packages/protocol/contracts/governance/Governance.sol

```diff
@@ -657,20 +662,30 @@ contract Governance is
   function revokeVotes() external nonReentrant returns (bool) {
     address account = getAccounts().voteSignerToAccount(msg.sender);
     Voter storage voter = voters[account];
-    for (uint256 i = 0; i < dequeued.length; i = i.add(1)) {
-      VoteRecord storage voteRecord = voter.referendumVotes[i];
-      (Proposals.Proposal storage proposal, Proposals.Stage stage) =
-        requireDequeuedAndDeleteExpired(
-          voteRecord.proposalId,
-          i
-        );
+    for (
+      uint256 dequeueIndex = 0;
+      dequeueIndex < dequeued.length;
+      dequeueIndex = dequeueIndex.add(1)
+    ) {
+      VoteRecord storage voteRecord = voter.referendumVotes[dequeueIndex];
+      (Proposals.Proposal storage proposal, Proposals.Stage stage) = requireDequeuedAndDeleteExpired(
+        voteRecord.proposalId,
+        dequeueIndex
+      );
       require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
-      proposal.updateVote(voteRecord.weight, 0, voteRecord.value, Proposals.VoteValue.None);
-      proposal.networkWeight = proposal.networkWeight - voteRecord.weight;
-      voter.referendumVotes[i] = VoteRecord(Proposals.VoteValue.None, voteRecord.proposalId, 0);
+      uint256 weight = voteRecord.weight;
+      Proposals.VoteValue value = voteRecord.value;
+      proposal.updateVote(weight, 0, value, Proposals.VoteValue.None);
+      proposal.networkWeight = getLockedGold().getTotalLockedGold();
+      voter.referendumVotes[dequeueIndex] = VoteRecord(
+        Proposals.VoteValue.None,
+        voteRecord.proposalId,
+        0
+      );
```

Comment:
we can just remove this because of the `delete`
```suggestion
```

Reply:
I should have noticed this. Thank you.